web/templates/home.html

```diff
@@ -1,6 +1,50 @@
 {% extends 'base.html' %}
 {% block content %}
-<div>
-    Test
-</div>
+        <div class="container">
+
+            {% for chart in chart_types %}
+                <div class="card bg-dark" style="min-width: 75vw;">
+                    <div class="card-header">{{ _(translations[chart]['bare']) }}</div>
+                    <div class="card-body">
+                        <div class="chart-plotly" id="chart-{{chart}}"></div>
+                    </div>
+                </div>
+                <br>
+            {% endfor %}
+        </div>
+
+        <script>
+            // this will run every second to update the charts
+            var update_charts = () => {
+                // user interaction will mutate layout and set autorange to false, so we need to reset it to true
+
+                $.ajax({
+                    url: "/callback/dashboard",
+                    type: "GET",
+                    contentType: 'application/json;charset=UTF-8',
+                    data: {
+                    },
+                    dataType:"json",
+                    success: function (data) {
+                        for(var i in data) {
+                            Plotly.react(data[i].layout.meta.id, data[i].data, data[i].layout, data[i].config).then(function(chart) {
+                                resizeObserver.observe(chart);
+                            });
+                        }
+                    }
+                });
+            };
+
+            // setup update timer
+            var timer = setInterval(update_charts, 2000);
```

Comment:
> I think you meant to use `setTimeout` here instead of `setInterval`. `setInterval` will call the given function every 2 seconds, but you're setting that up every time you call the function. If you're going to do a recursive callback like this you should use `setTimeout` since that just does the recursive call once instead of setting up a recurring event.
> 
> Also, you should probably call `setTimeout` at the end of your ajax success function in case getting the data from the server or displaying it takes more than two seconds.

I want the chart to update with new data every 2 seconds. Using the `setTimeout` I just get static charts. Am I missing something?